lib/autocomplete-ruby-client.coffee

```diff
@@ -1,61 +1,105 @@
-    # This means that running two atoms and closing one, kills rsense for the other
-    return if @rsenseStarted == true
-    @rsenseStarted = true
+    exec("ps -ef | head -1; ps -ef | grep java",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error looking for resense process',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @rsenseProcess = $.grep(TableParser.parse(stdout), (process) ->
+            process.CMD.join(' ').match( /rsense.*--port.*--path/ )
+          )[0]
+          if @rsenseProcess == undefined || @rsenseProcess == null
+            start()
+          else
+            @rsenseStarted = true
+    )
+
+  # Before trying to start in Windows we need to kill any existing rsense servers, so
+  # as to not end up with multiple rsense servsers unkillable by 'rsense stop'
+  # This means that running two atoms and closing one, kills rsense for the other
+  startRsenseWin32: =>
+    return if @rsenseStarted
+    start = @startRsenseCommand
 
     exec("#{@rsensePath} stop",
       (error, stdout, stderr) =>
         if error == null
-
-          port = atom.config.get('autocomplete-ruby.port')
-          exec("#{@rsensePath} start --port #{port} --path #{@projectPath}",
-            (error, stdout, stderr) ->
-              if error != null
-                atom.notifications.addError('Error starting rsense',
-                    {detail: "exec error: #{error}", dismissable: true}
-                  )
-                @rsenseStarted = false
-          )
-
+          start()
         else
           atom.notifications.addError('Error stopping rsense',
               {detail: "exec error: #{error}", dismissable: true}
             )
           @rsenseStarted = false
     )
 
-  stopRsense: ->
+  startRsenseCommand: =>
+    return if @rsenseStarted
+    exec("#{@rsensePath} start --port #{@port} --path #{@projectPath}",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error starting rsense',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @rsenseStarted = true
+    )
+
+  # First count how many atom windows are open.
+  # If there is only one open, then kill the rsense process.
+  # This is also able to kill an rsense process without a pid file.
+  # Otherwise do nothing so you will still be able to use rsense in other windows.
+  stopRsenseUnix: =>
+    stopCommand = @stopRsense
+
+    exec("ps -ef | head -1; ps -ef | grep atom",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error looking for atom process',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @atomProcesses = $.grep(TableParser.parse(stdout), (process) ->
+            process.CMD.join(' ').match( /--type=renderer.*--node-integration=true/ )
```

Comment:
Missed that, LGTM :+1:

I'll get this merged and released, looks like it should work in Windows if rsense is working.